src/ast/passes/resource_analyser.cpp

```diff
@@ -417,6 +476,12 @@ bool ResourceAnalyser::exceeds_stack_limit(size_t size)
   return size > bpftrace_.config_.get(ConfigKeyInt::on_stack_limit);
 }
 
+bool ResourceAnalyser::map_key_exceeds_stack_limit(const Map &map)
+{
+  return get_map_key_size(map) >
+         bpftrace_.config_.get(ConfigKeyInt::on_stack_limit);
```

Comment:
nit: can you use your other helper?
```suggestion
  return exceeds_stack_limit(get_map_key_size(map));
```